src/job_queue/job_result_wrapper.rs

```diff
@@ -0,0 +1,267 @@
+///
+/// # Type Parameters
+///
+/// * `T`: The specific type of the job result being wrapped. This type must be
+///   `'static`, `Send`, and `Sync`.
+///
+/// `JobResultWrapper` also implements the following traits if T
+/// implements the trait:
+///   Debug, Clone, Copy, Display, PartialOrd, Ord, PartialEq, Eq,
+pub struct JobResultWrapper<T>(T);
+
+impl<T: 'static + Send + Sync> JobResult for JobResultWrapper<T> {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn into_any(self: Box<Self>) -> Box<dyn Any> {
+        self
+    }
+}
+
+impl<T> Deref for JobResultWrapper<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T> DerefMut for JobResultWrapper<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl<T: 'static> TryFrom<JobCompletion> for JobResultWrapper<T> {
+    type Error = JobHandleError;
+
+    fn try_from(job_completion: JobCompletion) -> Result<Self, Self::Error> {
+        JobResultWrapper::try_from_completion(job_completion)
+    }
+}
+
+impl<'a, T: 'static> TryFrom<&'a JobCompletion> for &'a JobResultWrapper<T> {
+    type Error = JobHandleError;
+
+    fn try_from(job_completion: &'a JobCompletion) -> Result<Self, Self::Error> {
+        JobResultWrapper::try_from_completion_ref(job_completion)
+    }
+}
+
+impl<T: 'static> TryFrom<Box<dyn JobResult>> for JobResultWrapper<T> {
+    type Error = JobHandleError;
+
+    fn try_from(job_result: Box<dyn JobResult>) -> Result<Self, Self::Error> {
+        JobResultWrapper::try_from_boxed_job_result(job_result)
+    }
+}
+
+impl<'a, T: 'static> TryFrom<&'a dyn JobResult> for &'a JobResultWrapper<T> {
+    type Error = JobHandleError;
+
+    fn try_from(job_result: &'a dyn JobResult) -> Result<Self, Self::Error> {
+        JobResultWrapper::try_from_boxed_job_result_ref(job_result)
+    }
+}
+
+impl<T: 'static + Send + Sync> From<JobResultWrapper<T>> for Box<dyn JobResult> {
+    fn from(wrapper: JobResultWrapper<T>) -> Self {
+        Box::new(wrapper) as Box<dyn JobResult>
+    }
+}
+
+impl<T: Debug> Debug for JobResultWrapper<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("JobResultWrapper").field(&self.0).finish()
+    }
+}
+
+impl<T: Display> Display for JobResultWrapper<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+// Clippy warns about a manual Clone impl on a Copy type, but this is necessary
+// to handle the case where T is Clone but not Copy.
+//
+// Ideally, this could be expressed more clearly using negative trait bounds
+// (#[feature(negative_impls)]), like this:
+//
+// impl<T: Clone + !Copy> Clone for JobResultWrapper<T> {..}
+// impl<T: Clone + Copy> Clone for JobResultWrapper<T> {..}
+//
+// However, negative trait bounds are not yet stable in Rust as of rustc
+// v1.86.0.
+#[allow(clippy::expl_impl_clone_on_copy)]
```

Comment:
I strongly suggest to use `expect` over `allow` wherever possible.

Reply:
no longer relevant as of 8b189964d2c0fd51b8e9c4f7899b2f713bc95b99.  trait is derived instead.